conftest.py

```diff
@@ -152,6 +154,25 @@ class mscolab_settings(object):
 sys.path.insert(0, parent_path)
 
 
+@pytest.fixture(autouse=True)
+def close_open_windows(request):
+    """
+    Closes all windows after every test
+    """
+    # Mock every MessageBox widget in the test suite to avoid unwanted freezes on unhandled error popups etc.
+    with mock.patch("PyQt5.QtWidgets.QMessageBox.question"), mock.patch("PyQt5.QtWidgets.QMessageBox.information"), \
+            mock.patch("PyQt5.QtWidgets.QMessageBox.critical"), mock.patch("PyQt5.QtWidgets.QMessageBox.warning"):
+        yield
+
+    # Try to close all remaining widgets after each test
+    for qobject in set(QtWidgets.QApplication.topLevelWindows() + QtWidgets.QApplication.topLevelWidgets()):
+        try:
+            qobject.destroy()
+        # Some objects deny permission, pass in that case
+        except RuntimeError:
+            pass
```

Comment:
Should we act on an *unexpected* message window?

Reply:
I am not sure what you mean here.
Should we do more than pass on RuntimeError?
Windows/Widgets which throw a RuntimeError on destroyed seem to be not persistent. So ignoring them will usually solve the issue and leave no window open anyway.

Should we act on unexpected message windows? Absolutely. Message windows block pytest, any test which unexpectedly cause a message window to pop up without handling it is an infinite freeze.

Reply:
question: what happens when now an unexpected message window appears?
The test should fail, shouldn't it?

Reply:
Previously it froze, so now it's obviously much better already.